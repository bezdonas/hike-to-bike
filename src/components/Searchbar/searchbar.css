.searchbar {
  line-height: 20px;
  font-size: 16px;
  width: 150px;
}
