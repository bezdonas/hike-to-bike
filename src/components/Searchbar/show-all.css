.show-all-wrapper {
  margin-top: 10px;
  text-align: left;
  color: rgba(0, 0, 0, 0.8);
  font-size: 12px;
}

.show-all-wrapper input {
  display: inline-block;
  vertical-align: middle;
}
