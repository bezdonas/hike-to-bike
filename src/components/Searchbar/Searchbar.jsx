import { h, Component } from 'preact';
import './searchbar.css';

// SearchBarProps = {
//  position: [lat, lng],
//  changePosition: function
// }
export default class Searchbar extends Component {
  constructor(props) {
    super(props);
    this.handleSearchChange = this.handleSearchChange.bind(this);
  }
  handleSearchChange(e) {
    this.props.changePosition(e.target.value);
  }
  render(props) {
    return (
      <input
        className="searchbar"
        type="text"
        value={props.position}
        onChange={this.handleSearchChange}
      />
    );
  }
}
