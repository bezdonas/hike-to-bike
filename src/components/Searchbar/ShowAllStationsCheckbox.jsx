import { h, Component } from 'preact';
import './show-all.css';

// ShowAllStationsCheckboxProps = {
//  showAllStations: boolean,
//  toggleShowAllStations: function,
// }
export default class ShowAllStationsCheckbox extends Component {
  render(props) {
    const { showAllStations, toggleShowAllStations } = this.props;
    return (
      <div class="show-all-wrapper">
        <label>
          Show all stations{' '}
          <input
            type="checkbox"
            onClick={toggleShowAllStations}
            value={showAllStations}
          />
        </label>
      </div>
    );
  }
}
