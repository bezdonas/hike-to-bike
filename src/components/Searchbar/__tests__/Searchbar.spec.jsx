import { h, preact } from 'preact';
import render from 'preact-render-to-json';
import Searchbar from '../Searchbar.jsx';

describe('Searchbar', () => {
  it('renders as expected and matches snapshot', () => {
    const props = {
      position: [1, 1],
      changePosition: () => false,
    };
    expect(render(<Searchbar {...props} />)).toMatchSnapshot();
  });
});
