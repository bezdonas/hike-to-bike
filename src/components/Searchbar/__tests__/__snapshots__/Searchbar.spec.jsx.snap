// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar renders as expected and matches snapshot 1`] = `
<input
  class="searchbar"
  onChange={[Function]}
  type="text"
  value={
    Array [
      1,
      1,
    ]
  }
/>
`;
